### ISSUE TITLE
garbage collector seems not work

### ISSUE THREAD

**Issue by jcbritobr** [diamondburned/gtkcord4#143]
I'm using gtkcord and its working very well, but notice that the resources allocated keep going allocated even when you got a repeated page, and they aren't collected.
Memory got high easy only changing pages, and even the same pages.

![image](https://github.com/diamondburned/gtkcord4/assets/6579862/7bd5fa0d-7103-441b-abe1-1dfe0d7aede8)

**Comment by diamondburned** [comment]
You're absolutely correct. gtkcord4 has had a memory leak since the start; it mostly has to do with the internal weak reference code breaking and I don't have any way of debugging why that happens.

If anyone could fix this on gotk4's side, that would be awesome. I think the new `runtime.Pinner` API will help a lot but that would require refactoring the internal code and I haven't had the time to do that yet.

**Comment by gschurck** [comment]
Memory usage seems to go very high also when opening a channel with many images.

**Comment by jgardona** [comment]
Hello, I'm here again, as golang 1.24 got a weak reference. Maybe this feature can solve this issue.

**Comment by diamondburned** [comment]
https://github.com/diamondburned/gotk4/pull/161 actually exists for this, but Dissent still crashes with this branch, so there's some kinks to work out still.
